main.js

```diff
@@ -0,0 +1,379 @@
+    };
+    if (playerOneColor != null && playerTwoColor != null && playerThreeColor != null && toggleAICount === 0 || playerOneColor != null && playerTwoColor != null && toggleAICount === 1 || playerOneColor != null && playerTwoColor != null && playerModeToggle === 1) {
+        $('.choose-color-page').hide();
+        $('.game_area').show();
+        $('.gameTitle').text("Player One's Turn");
+        playerSwitch = 2;
+    }
+}
+
+function aiSelectColor() { // Allows AI to pick random color after player one chooses color
+    var colorArray = [$(".red").css("background-image"), $(".blue").css("background-image"), $(".green").css("background-image"), $(".green").css("background-image")];
+    if (playerOneColor != null) {
+        for (var colorCount = 0; colorCount < colorArray.length; colorCount++) {
+            if (playerOneColor === colorArray[colorCount]) {
+                colorArray.splice(colorCount, 1); //removes player one color from colorArray so AI doesn't pick undefined color
+            }
+        }
+        var randomColorNum = Math.floor((Math.random() * colorArray.length ));
+        playerTwoColor = colorArray[randomColorNum];
+    }
+    toggleAISwitch = 1 - toggleAISwitch;
+};
+
+function addToGridArray() { //function that stores respective player's number into gameArray
+    for (var rowCount = gameArray.length-1; rowCount >= 0; rowCount--) {
+        if (gameArray[rowCount][columnNumber] === null) {
+            if (playerSwitch === 2) {
+                gameArray[rowCount][columnNumber] = 2;
+                break;
+            } else if (playerSwitch === 1) {
+                gameArray[rowCount][columnNumber] = 1;
+                break;
+            } else if (playerSwitch === 0 && playerModeToggle === 0) {
+                gameArray[rowCount][columnNumber] = 0;
+                break;
+            }
+        }
+    }
+    playerSwitch--;
+    checkPowerUpCondition();
+    if (playerSwitch === 1){
+        $('.gameTitle').text("Player Two's Turn");
+    } else if (playerSwitch === 0 && playerModeToggle === 0){
+        $('.gameTitle').text("Player Three's Turn");
+    }
+    addColorToGrid();
+    checkWinCondition();
+    if (toggleAICount === 1 && playerSwitch === 1) { //if AI mode is on, runs function that has AI "choose" a column
+        aiGridSelect();
+    }
+    if (playerSwitch === -1 || playerSwitch === 0 && toggleAICount === 1 || playerSwitch === 0 && playerModeToggle === 1) {
+        playerSwitch = 2;
+        $('.gameTitle').text("Player One's Turn");
+    }
+}
+
+function aiGridSelect() { //function that allows AI to randomly select a column
+    var randomColumnNum = Math.floor((Math.random() * 6 ));
+    columnNumber = randomColumnNum;
+    $('.gameTitle').text("Turn of the Gods");
+    setTimeout(addToGridArray, 2000);
+}
+
+
+function addColorToGrid() { //function that adds player's image to HTML grid depending on gameArray
+    for (var rowCount = gameArray.length-1; rowCount >= 0; rowCount--) {
```

Comment:
Same situation here. There is nothing to account for a column being fully occupied.